plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/types/TypesGeneratorFragment2.xtend

```diff
@@ -12,10 +12,20 @@ import org.eclipse.xtext.xtext.generator.AbstractGeneratorFragment2
 import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess
 
 import static extension org.eclipse.xtext.xtext.generator.model.TypeReference.*
+import org.eclipse.xtext.xtext.generator.xbase.XbaseUsageDetector
+import com.google.inject.Inject
+import org.eclipse.xtend.lib.annotations.Accessors
 
 class TypesGeneratorFragment2 extends AbstractGeneratorFragment2 {
 	
+	@Inject XbaseUsageDetector xbaseUsageDetector
+	
+	@Accessors boolean onlyEnabledIfGrammarIsUsed = false
```

Comment:
Yes, there are more accurate solutions. I went for a quick and 'not' dirty one :-)
Let's go with this and improve later.